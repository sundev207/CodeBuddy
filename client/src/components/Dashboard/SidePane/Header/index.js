import React from 'react';
import { MessagesIcon, EventsIcon } from '../../../../assets/icons.js';
import { connect } from 'react-redux';

export const DashHeaderContent = ({currentUser}) => (
  currentUser !== undefined &&
  <div className='dash-header'>
    <h1 id='dash-header-title'>CodeBuddy</h1>
    <div className='dash-header-content'>
      <div className='search-bar'>
        <input type='text' />
      </div>
      <MessagesIcon />
      <EventsIcon />
      <p>{currentUser.profile.userName}</p>
      <img id='header-img' src={currentUser.profile.img} alt='' />
    </div>
  </div>
);

const mapStateToProps = state => {
  return {
    currentUser: state.user[0]
  }
}

export const DashHeader = connect(mapStateToProps)(DashHeaderContent)