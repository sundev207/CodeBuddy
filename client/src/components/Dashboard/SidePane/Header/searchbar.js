import React from 'react';
import Select from 'react-select';

const SearchBar = () => (
  <div className='search-bar'>
    <input type='text' />
  </div>
)