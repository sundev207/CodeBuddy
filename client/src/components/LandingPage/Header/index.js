import React, { Component } from 'react';
import { MenuIcon } from '../../../assets/icons';

const ListMenu = () => (
  <ul className="nav">
    <li><a href="/">Home</a></li>
    <li><a href="/">Contact</a></li>
    <li><a href="/">About</a></li>
  </ul>
);

const MobileListMenu = ({ onClick, menuActive }) => (
  <ul className={`mobile-nav expanded-${menuActive}`}>
    {!menuActive
      ? <MenuIcon onClick={onClick} />
      : <ListMenu />
    }
  </ul>
);

export class Header extends Component {
  state = {
    width: null,
    menuActive: false,
  };

  componentDidMount() {
    this.updateWindowDimensions();
    window.onresize = () => this.updateWindowDimensions();
  }

  updateWindowDimensions = () => {
    const width = window.innerWidth
      || document.documentElement.clientWidth
      || document.body.clientWidth;

    this.setState({ width });
  }

  toggleMenu = () => this.setState((prevState, props) => ({
    menuActive: !prevState.menuActive,
  }));

  render() {
    const { menuActive } = this.state;
    return (
      <div className="header">
        <h1>CodeBuddy</h1>
        {this.state.width > 700
          ? <ListMenu />
          : <MobileListMenu onClick={this.toggleMenu} menuActive={menuActive} />
        }
      </div>
    );
  }
}
