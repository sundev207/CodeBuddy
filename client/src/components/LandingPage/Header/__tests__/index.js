import React from 'react';
import { render, cleanup } from 'react-testing-library';
import Header from '..';

describe('Header component', () => {
  afterEach(cleanup);

  it('renders correctly', () => {
    expect(render(<Header />).container).toMatchSnapshot();
  });
});
