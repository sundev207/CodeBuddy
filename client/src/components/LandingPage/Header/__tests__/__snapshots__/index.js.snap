// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly 1`] = `
<div>
  <div
    class="header"
  >
    <h1>
      CodeBuddy
    </h1>
    <ul
      class="nav"
    >
      <li>
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/"
        >
          Contact
        </a>
      </li>
      <li>
        <a
          href="/"
        >
          About
        </a>
      </li>
    </ul>
  </div>
</div>
`;
